content/03-reference/01-tools-and-interfaces/02-prisma-client/078-error-reference.mdx

```diff
@@ -0,0 +1,92 @@
+---
+title: 'Prisma Client error reference'
+metaTitle: 'Prisma Client error reference (Reference)'
+metaDescription: 'Prisma Client errors'
+---
+
+<TopBlock>This page describes Prisma Client exceptions and error codes.</TopBlock>
+
+## <inlinecode>PrismaClientKnownRequestError</inlinecode>
+
+Prisma Client throws a `PrismaClientKnownRequestError` exception if the query engine returns a known error related to the request - for example, a unique constraint violation.
+
+| **Property** | **Description**                                                                                                  |
+| :----------- | :--------------------------------------------------------------------------------------------------------------- |
+| `code`       | A Prisma-specific [error code](#error-codes).                                                                    |
+| `meta`       | Additional information about the error - for example, the field that caused the error: `{ target: [ 'email' ] }` |
+
+### Error codes
+
+#### Query Engine
+
+| **Error** | **Example message**                                                                                                                                           | **Description** |
+| :-------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------ | :-------------- |
+| `P2000`   | "The provided value for the column is too long for the column's type. Column: \${column_name}"<br />                                                          |
+| `P2001`   | "The record searched for in the where condition (`${model_name}.${argument_name} = ${argument_value}`) does not exist"<br />                                  |
+| `P2003`   | "Foreign key constraint failed on the field: `${field_name}`"<br />                                                                                           |
```

Comment:
Why are
P2002 : UniqueKeyViolation
P2004 : ConstraintViolation
P2007 : TypeMismatchInvalidCustomType
P2009 : QueryValidationFailed
P2010 : RawQueryFailed
P2011 : NullConstraintViolation
P2012 : MissingRequiredValue
P2015 : RelatedRecordNotFound
P2016 : InterpretationError
P2019 : Input Error from 
P2020 : ValueOutOfRange
not included?